src/kernels/kernelFinder.base.ts

```diff
@@ -172,35 +177,56 @@ export abstract class BaseKernelFinder implements IKernelFinder {
     ) {
         const kernelsFromCachePromise =
             useCache === 'ignoreCache' ? Promise.resolve([]) : this.getFromCache(kind, cancelToken);
-        let kernelsRetrievedFromCache: boolean | undefined;
+        let updateCache = true;
         const kernelsWithoutCachePromise = finder();
         let kernels: KernelConnectionMetadata[] = [];
         if (useCache === 'ignoreCache') {
-            kernels = await kernelsWithoutCachePromise;
+            try {
+                kernels = await kernelsWithoutCachePromise;
+            } catch (ex) {
+                // Since fetching the remote kernels failed, we fall back to the cache,
+                // at this point no need to display all of the kernel specs,
+                // Its possible the connection is dead, just display the live kernels we had.
+                // I.e. if user had a notebook connected to a remote kernel, then just display that live kernel.
+                kernels = await this.getFromCache(kind, cancelToken);
```

Comment:
This is what makes #9167 possible, display the old dead kernels and notify the user and let the user remove it from the list of kernels.